File: Java/Combustivel.java
public class Combustivel {
    // Final: Não pode ser alterado
    // Static: Pertence a classe, não ao objeto
    private final static double PRECO_ETANOL_LITRO = 2.27;
    private final static double PRECO_GASOLINA_LITRO = 2.90;
    private final TipoCombustivel tipoCombustivel;
    private final double precoLitro;

    public Combustivel(TipoCombustivel tipoCombustivel) {
        this.tipoCombustivel = tipoCombustivel;
        this.precoLitro = this.getPrecoLitro();
    }

    public double getPrecoLitro() {
        if (tipoCombustivel == TipoCombustivel.ETANOL)
            return PRECO_ETANOL_LITRO;
        else
            return PRECO_GASOLINA_LITRO;
    }

    public TipoCombustivel getTipoCombustivel() {
        return tipoCombustivel;
    }

    @Override
    public String toString() {
        return "Combustivel {\n" +
                "\ttipoCombustivel = " + tipoCombustivel + ",\n" +
                "\tprecoLitro = " + precoLitro + "\n" +
                '}';
    }
}

File: Python/Combustivel.py
from enum import Enum


class Combustivel:
    PRECO_ETANOL_LITRO = 2.27
    PRECO_GASOLINA_LITRO = 2.90

    def __init__(self, tipo_combustivel):
        self.tipo_combustivel = tipo_combustivel
        self.preco_litro = self.get_preco_litro()

    def get_preco_litro(self):
        if self.tipo_combustivel == TipoCombustivel.ETANOL:
            return Combustivel.PRECO_ETANOL_LITRO
        else:
            return Combustivel.PRECO_GASOLINA_LITRO

    def get_tipo_combustivel(self):
        return self.tipo_combustivel

    def __str__(self):
        return f"Combustivel {{\n\ttipo_combustivel = {self.tipo_combustivel},\n\tpreco_litro = {self.preco_litro}\n}}"


class TipoCombustivel(Enum):
    GASOLINA = 1
    ETANOL = 2
